### ISSUE TITLE
crash probably cause of float/double code (?)

### ISSUE THREAD

**Issue by alexalkis** [bebbo/gcc#25]
clone this https://github.com/glouw/dungen
At arround line 292 on Map.c, change to:

```
static float mag(const Point a)
{
#ifdef __AMIGA
  return sqrt(a.x * a.x + a.y * a.y);
#else
  return sqrtf(a.x * a.x + a.y * a.y);
#endif
}

```
make CC=m68k-amigaos-gcc CFLAGS=-noixemul

Executable crashes on (emulated) A1200 (8000 000B)

**Comment by MBeijer** [comment]
try adding -msoft-float to your CFLAGS :)

**Comment by alexalkis** [comment]
No go for me with -msoft-float.  Still crashing with 8000 000B.
Does it work for you?

**Comment by bebbo** [comment]
it's working for me - usind -noixemul - but I have already the fix from ticket #21.

**Comment by alexalkis** [comment]
Updated to the latest toolchain.  Still no go for me.  Can a kind soul check the attached executable if it works or crashes, please?
[dungen.zip](https://github.com/bebbo/gcc/files/1745244/dungen.zip)

**Comment by bebbo** [comment]
runs and prints a maze

**Comment by alexalkis** [comment]
Huh? It was the 68881.
Was there an architecture change?  Do floats need fpu now?  
How can I compile for "stock A1200"?  (I've tried -msoft-float, still crashes with no fpu)

**Comment by bebbo** [comment]
As I wrote: It runs on a plain 68k also on a plain 68020.

But you must not use math libraries which require a 68881!

**Comment by alexalkis** [comment]
Just to be clear .... :)

When you say "it runs on a plain 68k also on a plain 68020" you are referring to the executable I attached a few posts above?  (I am trying to understand if I have a broken amiga installation or not)

**Comment by bebbo** [comment]
Yes, it's your executable - which is almost bit compatible to mine (need to update the libnix libs).
I'am testing on a fresh installed
* kickstart 40.63
* workbench 40.42
```
list libs:m#?
Directory "libs:" on Wednesday 21-Feb-18
mathtrans.library           4280 ----rwed 06-Jul-94 11:38:58
mathieeesingtrans.library   10428 ----rwed 06-Jul-94 11:38:58
mathieeesingbas.library     4384 ----rwed 04-Jul-16 21:06:27
: From: Kickstart 40.63 (A500/A600/A2000)
mathffp.library             1244 ----rwed 04-Jul-16 21:06:27
: From: Kickstart 40.63 (A500/A600/A2000)
mathieeedoubbas.library     5240 ----rwed 06-Jul-94 11:38:58
mathieeedoubtrans.library   16584 ----rwed 06-Jul-94 11:38:58
6 files - 92 blocks used 
```

**Comment by alexalkis** [comment]
Interesting, I am missing mathieeesingbas.library and mathffp.library.

**Comment by bebbo** [comment]
[Libs.zip](https://github.com/bebbo/gcc/files/1745444/Libs.zip)

**Comment by bebbo** [comment]
[Libs.zip](https://github.com/bebbo/gcc/files/1745445/Libs.zip)

**Comment by alexalkis** [comment]
Thanks for the libs. I copied them.  Still crashes without FPU.
Kickstart 40.68
Workbench 40.42
FS-UAE 2.8.3 (Built for Linux x86-64)
UAE: KS ROM v3.1 (A1200) rev 40.68 (512k)

It doesn't make sense or is it me? I mean, why it runs ok with FPU?

**Comment by alexalkis** [comment]
snoopdos says it opens
mathieeedoubtrans.library
mathieeesingbas.library
mathieeedoubbas.library
utility.library
all ok

**Comment by alexalkis** [comment]
I get a failure in snoopdos
Process Name: ramlib 
Action: OpenRes
Target Name: MathIEEE.resource
Don't know if it's relevant or normal failure.

**Comment by alexalkis** [comment]
Anyways, since my executable works on a system with no fpu there is no point of keeping this issue open.  It's obviously something on my side.  I'll sort it out, hopefully! :)

**Comment by bebbo** [comment]
heh! I can get a Guru 8100 0005 from that program, when it exits after printing the maze.
But that means there is a corrupted memory list at exit -> the program has an error.

**Comment by alexalkis** [comment]
Hmmm, valgrind on linux doesn't complain about anything :)

==11363== 
==11363== HEAP SUMMARY:
==11363==     in use at exit: 0 bytes in 0 blocks
==11363==   total heap usage: 603 allocs, 603 frees, 6,902,664 bytes allocated
==11363== 
==11363== All heap blocks were freed -- no leaks are possible
==11363== 
==11363== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
==11363== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)

**Comment by matthey7** [comment]
> Interesting, I am missing mathieeesingbas.library and mathffp.library.

The mathieeesingbas.library and mathffp.library are in kickstart (ROM or kickstart flash). They should **not** normally be in SYS:Libs.

> I get a failure in snoopdos
> Process Name: ramlib
> Action: OpenRes
> Target Name: MathIEEE.resource

The AmigaOS 3.9 ieee math libraries for an FPU open the fpsp.resource. It looks like the MathIEEE.resource was used before the fpsp.resource FPU hook method as there is an old C= mathresource.h include which refers to it.

http://amigadev.elowar.com/read/ADCD_2.1/Includes_and_Autodocs_3._guide/node062C.html

You may have mismatched (old/new and/or FPU/no FPU) math libraries as bebbo already pointed out. Try installing the correct set of math libraries for your hardware and AmigaOS version.

**Comment by alexalkis** [comment]
@matthey7 thanks for the pointers.

I deleted all math#? from libs.  I booted of a workbench3.1 floppy image. I copied the math#? libs.
Executable crashes when rebooted from HD.

I booted from workbench2.04 floppy image. I copied the math#? libs.
Executable crashes when rebooted from HD.

I booted from workbench3.1 floppy image and then run the executable from HD (having floppy as SYS:), executable crashes.

This is so weird. Maybe it's kickstart.
